src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -1,26 +1,35 @@
-import React, { useCallback, useRef, useEffect } from 'react';
 import { extent } from 'd3-array';
 import { rgb } from 'd3-color';
-import { scaleLinear, scaleSequential } from 'd3-scale';
-import { Vector3 } from 'three';
-import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
-import { clamp, range } from 'lodash-es';
+import {
+  scaleSequential,
+  ScaleLinear,
+  ScaleSymLog,
+  scaleLinear,
+  scaleSymlog,
+} from 'd3-scale';
+import { range } from 'lodash-es';
 import { D3Interpolator } from './interpolators';
 
-const ZOOM_FACTOR = 0.95;
+export type ColorScale =
+  | ScaleLinear<number, number>
+  | ScaleSymLog<number, number>;
 
 export function computeTextureData(
   values: number[],
-  domain: [number, number] | undefined,
+  colorScale: ColorScale | undefined,
   interpolator: D3Interpolator
 ): Uint8Array | undefined {
-  if (domain === undefined) {
+  if (colorScale === undefined) {
     return undefined;
   }
-  const scale = scaleSequential(interpolator).domain(domain);
+
+  // Map colors to the output of colorScale
+  const scale = scaleSequential(interpolator).domain(
+    colorScale.range() as [number, number]
```

Comment:
It's a bit like doing `scale.copy()` in `ColorBar.tsx`. I'm not very fond of creating scales from existing scales... In this case, it forces you to cast the type of the range with `as`. I feel like each scale should be recreated from the raw building blocks (domain, range, etc.)

Don't worry, though, as my next PR will help with this.

Reply:
I would agree but in my case, I would need to pass around the type of scale, the domain and recreate the scale which seems heavy versus passing only the scale.

Wait and see :monocle_face: